connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java

```diff
@@ -0,0 +1,154 @@
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(RestServer.class)
+@PowerMockIgnore("javax.management.*")
+public class ConnectorPluginsResourceTest {
+
+    private static Map<String, String> props = new HashMap<>();
+    static {
+        props.put("test.string.config", "testString");
+        props.put("test.int.config", "10");
+    }
+
+    private static ConfigInfos CONFIG_INFOS;
+    static {
+        List<ConfigInfo> configs = new LinkedList<>();
+
+        ConfigKeyInfo configKeyInfo = new ConfigKeyInfo("test.string.config", "STRING", "", "HIGH", "Test configuration for string type.", null, -1, "NONE", "test.string.config", new LinkedList<String>());
+        ConfigValueInfo configValueInfo = new ConfigValueInfo("test.string.config", "testString", Collections.<Object>emptyList(), Collections.<String>emptyList(), true);
+        ConfigInfo configInfo = new ConfigInfo(configKeyInfo, configValueInfo);
+        configs.add(configInfo);
+
+        configKeyInfo = new ConfigKeyInfo("test.int.config", "INT", "", "MEDIUM", "Test configuration for integer type.", null, -1, "NONE", "test.int.config", new LinkedList<String>());
+        configValueInfo = new ConfigValueInfo("test.int.config", "10", Collections.<Object>emptyList(), Collections.<String>emptyList(), true);
+        configInfo = new ConfigInfo(configKeyInfo, configValueInfo);
+        configs.add(configInfo);
+
+        CONFIG_INFOS = new ConfigInfos(WorkerTestConnector.class.getName(), 0, Collections.<String>emptyList(), configs);
+    }
+
+    @Mock
+    private Herder herder;
+    private ConnectorPluginsResource connectorPluginsResource;
+
+    @Before
+    public void setUp() throws NoSuchMethodException {
+        PowerMock.mockStatic(RestServer.class,
+                             RestServer.class.getMethod("httpRequest", String.class, String.class, Object.class, TypeReference.class));
+        connectorPluginsResource = new ConnectorPluginsResource(herder);
+    }
+
+    @Test
+    public void testValidateConfig() throws Throwable {
+        herder.validateConfigs(EasyMock.eq(WorkerTestConnector.class.getName()), EasyMock.eq(props));
+
+        PowerMock.expectLastCall().andAnswer(new IAnswer<ConfigInfos>() {
+            @Override
+            public ConfigInfos answer() {
+                Config config = new WorkerTestConnector().validate(props);
+                return AbstractHerder.generateResult(WorkerTestConnector.class.getName(), config.configDef().configKeys(), config.configValues(), config.groups());
+            }
+        });
+
+        PowerMock.replayAll();
+
+        ConfigInfos configInfos = connectorPluginsResource.validateConfigs(WorkerTestConnector.class.getName(), props);
+        assertEquals(CONFIG_INFOS, configInfos);
+
+        PowerMock.verifyAll();
+    }
+
+    /* Name here needs to be unique as we are testing the aliasing mechanism */
+    public static class WorkerTestConnector extends Connector {
```

Comment:
Naming this the same as the one in `WorkerTest` is causing failures in `WorkerTest` because the search for the connector by reflection finds both classes.